beater/telemetry.go

```diff
@@ -0,0 +1,118 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package beater
+
+import (
+	"github.com/elastic/apm-server/beater/config"
+	"github.com/elastic/apm-server/idxmgmt"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/idxmgmt/ilm"
+	"github.com/elastic/beats/v7/libbeat/monitoring"
+)
+
+var apmRegistry = monitoring.GetNamespace("state").GetRegistry().NewRegistry("apm-server")
+var configRegistry = apmRegistry.NewRegistry("configuration")
```

Comment:
Can we do remove this "configuration" namespace? The libbeat state does not have such a namespace, and I'd like to be consistent if possible.

Reply:
consistent with what? Other beats derive registers from the state registry for telemetry too:

https://github.com/elastic/beats/blob/a6f180e273cb7de74a8ae5b704e85b076498a913/libbeat/cmd/instance/beat.go#L181
https://github.com/elastic/beats/blob/a6f180e273cb7de74a8ae5b704e85b076498a913/libbeat/monitoring/report/elasticsearch/elasticsearch.go#L357
etc

In this case, the reason is that if we later on add telemetry for other (dynamic) variables, we can separate them better and not have all counters mixed up at the same level.

Reply:
Consistent with the structure that libbeat produces:

```
    "management" : {
        "enabled" : false
    },
    "queue" : {
        "name" : "mem"
    },
    "output" : {
        "name" : "elasticsearch"
    },
```

Why can't we do the same, like this?

```
    "apm-server": {
        "rum": {
            "enabled": true
        },
        "kibana": {
            "enabled": true,
            "version": "7.8.0" (for example)
        },
        ...
    }
```